src/lib/platform/EiScreen.cpp

```diff
@@ -733,4 +775,40 @@ IKeyState* EiScreen::getKeyState() const
     return key_state_;
 }
 
+EiScreen::HotKeyItem::HotKeyItem(std::uint32_t mask, std::uint32_t id) :
+   mask_(mask),
+   id_(id)
+{
+}
+
+EiScreen::HotKeySet::HotKeySet(KeyID key) :
+  id_(key)
+{
+}
+
+bool EiScreen::HotKeySet::remove_by_id(std::uint32_t id)
+{
+    for (auto item = set_.begin(); item != set_.end(); ++item) {
+        if (item->id_ == id) {
+            set_.erase(item);
+            return true;
+        }
+    }
+    return false;
+}
+
+void EiScreen::HotKeySet::add_item(HotKeyItem item)
+{
+    set_.push_back(item);
+}
+
+std::uint32_t EiScreen::HotKeySet::find_by_mask(std::uint32_t mask) const
+{
+    for (auto item = set_.begin(); item != set_.end(); ++item) {
```

Comment:
any reason you can't use a `ranged for` here ? (and else where w/ `set_` ? )

Reply:
Fixed.